corehq/apps/locations/models.py

```diff
@@ -611,6 +619,23 @@ def sql_location(self):
         notify_of_deprecation("'sql_location' was just called on a sql_location.  That's kinda silly.")
         return self
 
+    @classmethod
+    def location_and_descendants(cls, location_ids):
+        locations = []
+        for location_id in location_ids:
+            location = cls.by_location_id(location_id)
```

Comment:
It would be interesting to see in which case it would not find a location. It would raise an exception at next step anyway but `DoesNotExist` would make it explicit.
Would use `get` instead of `by_location_id`